test/test_sort.py

```diff
@@ -469,6 +469,69 @@ def test_case_sensitive_only_affects_text(self):
         self.assertEqual(results[-1].track, 10)
 
 
+class NonExistingFieldTest(DummyDataTestCase):
+    """Test sorting by non-existing fields"""
+
+    def test_non_existing_fields_not_fail(self):
+        qs = ['foo+', 'foo-', '--', '-+', '+-', '++', '-foo-', '-foo+', '---']
+
+        q0 = 'foo+'
+        results0 = list(self.lib.items(q0))
+        for q1 in qs:
+            results1 = list(self.lib.items(q1))
+            for r1, r2 in zip(results0, results1):
+                self.assertEqual(r1.id, r2.id)
+
+    def test_combined_non_existing_field_asc(self):
+        all_results = list(self.lib.items('id+'))
+        q = 'foo+ id+'
+        results = list(self.lib.items(q))
+        self.assertEqual(len(all_results), len(results))
+        for r1, r2 in zip(all_results, results):
+            self.assertEqual(r1.id, r2.id)
+
+    def test_combined_non_existing_field_desc(self):
+        all_results = list(self.lib.items('id+'))
+        q = 'foo- id+'
+        results = list(self.lib.items(q))
+        self.assertEqual(len(all_results), len(results))
+        for r1, r2 in zip(all_results, results):
+            self.assertEqual(r1.id, r2.id)
+
+    def test_field_present_in_some_items(self):
```

Comment:
A small remark for this test case: when the field is only present on some items and sorting by that field on *ascending* order, the items that do not have that field are displayed at the *top* of the list. While it does make sense, I'm wondering if it might be a bit confusing for end users: perhaps a small clarification or note on the `Queries - Sort order` docs might be in order?